### ISSUE TITLE
Why is decoder_input of size 3 ?

### ISSUE THREAD

**Issue by ashishkr23** [FGiuliari/Trajectory-Transformer#17]
Hi

Thanks for the code and the paper. Awesome work :)

While initialising the individualTF model why does the decoder_input/decoder_output have size 3 whereas the the encoder_input size is 2. Hence the ouput of the decoder is of the shape [N,Steps,3]. What is the third value we are trying to predict per step ? I was assuming we are trying to predict only the relative displacements in the cartesian coordinates ?
Also what does the extra term in the loss signify ? `torch.mean(torch.abs(pred[:,:,2])`

Would be great if you could clarify this doubt.

Thanks

**Comment by BYZANTINE26** [comment]
> Hi
> 
> Thanks for the code and the paper. Awesome work :)
> 
> While initialising the individualTF model why does the decoder_input/decoder_output have size 3 whereas the the encoder_input size is 2. Hence the ouput of the decoder is of the shape [N,Steps,3]. What is the third value we are trying to predict per step ? I was assuming we are trying to predict only the relative displacements in the cartesian coordinates ?
> Also what does the extra term in the loss signify ? `torch.mean(torch.abs(pred[:,:,2])`
> 
> Would be great if you could clarify this doubt.
> 
> Thanks

did you figured this out ?
